bundles/net.tourbook/src/net/tourbook/extension/export/ExportTourExtension.java

```diff
@@ -1,82 +1,92 @@
  * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110, USA
  *******************************************************************************/
 package net.tourbook.extension.export;
 
-import java.util.ArrayList;
+import java.util.List;
 
 import net.tourbook.data.TourData;
 
+import org.eclipse.jface.resource.ImageDescriptor;
+
 public abstract class ExportTourExtension {
 
-	private String	_exportId;
-	private String	_visibleName;
-	private String	_fileExtension;
-
-	/**
-	 * Exports the tour in the {@link TourData} list. If only one tour is exported, the values of
-	 * tourStartIndex and tourEndIndex is the range which points are exported, when the index is -1,
-	 * the whole tour is exported.
-	 * 
-	 * @param tourDataList
-	 * @param tourStartIndex
-	 * @param tourEndIndex
-	 */
-	public abstract void exportTours(ArrayList<TourData> tourDataList, int tourStartIndex, int tourEndIndex);
-
-	public String getExportId() {
-		return _exportId;
-	}
-
-	public String getFileExtension() {
-		return _fileExtension;
-	}
-
-	public String getVisibleName() {
-		return _visibleName;
-	}
-
-	public void setExportId(final String fExportId) {
-		this._exportId = fExportId;
-	}
-
-	public void setFileExtension(final String fFileExtension) {
-		this._fileExtension = fFileExtension;
-	}
-
-	public void setVisibleName(final String fVisibleName) {
-		this._visibleName = fVisibleName;
-	}
-
-	@Override
-	public String toString() {
-		return new StringBuilder()//
-				.append("id: ")//$NON-NLS-1$
-				.append(_exportId)
-				.append(" \t") //$NON-NLS-1$
-				//
-				.append("name: ") //$NON-NLS-1$
-				.append(_visibleName)
-				.append(" \t") //$NON-NLS-1$
-				//
-				.append("extension: ")//$NON-NLS-1$
-				.append(_fileExtension)
-				.append(" \t") //$NON-NLS-1$
-				//
-				.toString();
-
-	}
+   private String          _exportId;
+   private String          _visibleName;
+   private String          _fileExtension;
+   private ImageDescriptor _imageDescriptor;
+
+   /**
+    * Exports the tour in the {@link TourData} list. If only one tour is exported, the values of
+    * tourStartIndex and tourEndIndex is the range which points are exported, when the index is -1,
+    * the whole tour is exported.
+    *
+    * @param tourDataList
+    * @param tourStartIndex
+    * @param tourEndIndex
+    */
+   public abstract void exportTours(List<TourData> tourDataList, int tourStartIndex, int tourEndIndex);
+
+   public String getExportId() {
+      return _exportId;
+   }
+
+   public String getFileExtension() {
+      return _fileExtension;
+   }
+
+   public ImageDescriptor getImageDescriptor() {
+      return _imageDescriptor;
+   }
+
+   public String getVisibleName() {
+      return _visibleName;
+   }
+
+   public void setExportId(final String exportId) {
+      _exportId = exportId;
+   }
+
+   public void setFileExtension(final String fileExtension) {
+      _fileExtension = fileExtension;
+   }
+

```

Comment:
>I wouldn't have modernized it any other way 😄

Good to know !

Do you know why there was a f in front of the parameters ? was it a typo ?

`fVisibleName`, `fFileExtension` ....

Reply:
`f` was a prefix for `f`ields which is now `_` when it's private but there you can see how long I've not touched this code otherwise I would have changed it.